File: esp32/commands.py
"""
High level robot movement commands.
"""

from .esp32 import esp32


def forward():
    esp32.send("FORWARD")


def backward():
    esp32.send("BACKWARD")


def left():
    esp32.send("LEFT")


def right():
    esp32.send("RIGHT")


def stop():
    esp32.send("STOP")

File: firmware/commands.cpp
#include <Arduino.h>

#include "commands.h"
#include "motor.h"

void executeCommand(String command)
{
    command.toUpperCase();

    if (command == "FORWARD")
    {
        moveForward();
        Serial.println("OK FORWARD");
    }

    else if (command == "BACKWARD")
    {
        moveBackward();
        Serial.println("OK BACKWARD");
    }

    else if (command == "LEFT")
    {
        turnLeft();
        Serial.println("OK LEFT");
    }

    else if (command == "RIGHT")
    {
        turnRight();
        Serial.println("OK RIGHT");
    }

    else if (command == "STOP")
    {
        stopMotors();
        Serial.println("OK STOP");
    }

    else
    {
        Serial.println("UNKNOWN COMMAND");
    }
}